### ISSUE TITLE
Serialization Functionality

### ISSUE THREAD

**Issue by seanmor5** [elixir-nx/axon_onnx#3]
Checklist for serialization support of Axon models.

## Linear

- [x] `dense`
- [ ] `bilinear`
- [ ] `embedding`

## Convolutional

- [ ] `conv`
- [ ] `conv_transpose`
- [ ] `depthwise_conv`
- [ ] `separable_conv2d`
- [ ] `separable_conv3d`

## Dropout

- [ ] `dropout`
- [ ] `spatial_dropout`
- [ ] `alpha_dropout`
- [ ] `feature_alpha_dropout`

## Pooling

- [ ] `adaptive_avg_pool`
- [ ] `adaptive_max_pool`
- [ ] `adaptive_lp_pool`
- [ ] `avg_pool`
- [ ] `max_pool`
- [ ] `lp_pool`
- [ ] `global_avg_pool`
- [ ] `global_max_pool`
- [ ] `global_lp_pool`

## Normalization

- [ ] `batch_norm`
- [ ] `layer_norm`
- [ ] `instance_norm`
- [ ] `group_norm`

## Recurrent

- [ ] `conv_lstm`
- [ ] `lstm`
- [ ] `gru`

## Combinators

- [ ] `concatenate`
- [ ] `add`
- [ ] `subtract`
- [ ] `multiply`

## Activations

- [ ] `celu`
- [ ] `elu`
- [ ] `exp`
- [ ] `gelu`
- [ ] `hard_sigmoid`
- [ ] `hard_silu`
- [ ] `hard_tanh`
- [ ] `leaky_relu`
- [ ] `linear`
- [ ] `log_sigmoid`
- [ ] `mish`
- [ ] `relu`
- [ ] `relu6`
- [ ] `selu`
- [ ] `sigmoid`
- [ ] `silu`
- [ ] `softmax`
- [ ] `softplus`
- [ ] `softsign`
- [ ] `tanh`

## Special

- [ ] `constant`
- [ ] `nx`
- [ ] `layer`
